src/dialog/dlgkeywheel.cpp

```diff
@@ -0,0 +1,158 @@
+            ConfigKey("[Library]", "key_notation")));
+    m_notation = static_cast<KeyUtils::KeyNotation>(notation);
+    // Select a valid display and update
+    switchDisplay(0);
+}
+
+bool DlgKeywheel::eventFilter(QObject* obj, QEvent* event) {
+    if (event->type() == QEvent::KeyPress) {
+        // we handle TAB + Shift TAB to cycle through the display types
+        QKeyEvent* keyEvent = static_cast<QKeyEvent*>(event);
+        if (keyEvent->key() == Qt::Key_Tab) {
+            switchDisplay(+1);
+            return true;
+        } else if (keyEvent->key() == Qt::Key_Backtab) {
+            switchDisplay(-1);
+            return true;
+        }
+    } else if (event->type() == QEvent::MouseButtonPress) {
+        QMouseEvent* mouseEvent = static_cast<QMouseEvent*>(event);
+        // first button forward, other buttons backward cycle
+        switchDisplay(mouseEvent->button() == Qt::LeftButton ? +1 : -1);
+        return true;
+    }
+    // standard event processing
+    return QDialog::eventFilter(obj, event);
+}
+
+void DlgKeywheel::resizeEvent(QResizeEvent* ev) {
+    QSize newSize = ev->size();
+    int size = qMin(ui->graphic->size().height(), ui->graphic->size().width());
+    newSize = QSize(size, size);
+
+    ui->graphic->resize(newSize);
+}
+
+bool DlgKeywheel::isHiddenNotation(KeyUtils::KeyNotation notation) {
+    for (size_t i = 0; i < sizeof(kNotationHidden) / sizeof(kNotationHidden[0]); i++) {
+        if (kNotationHidden[i] == notation) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void DlgKeywheel::switchDisplay(int step) {
+    const int invalidNotations = static_cast<int>(KeyUtils::KeyNotation::Invalid);
+    const int numNotations = static_cast<int>(KeyUtils::KeyNotation::NumKeyNotations);
+
+    int newNotation = static_cast<int>(m_notation) + step;
+
+    if (step == 0) {
+        step = 1;
+    }
+    // we skip variants with redundant information
+    while (newNotation <= invalidNotations ||
+            newNotation >= numNotations ||
+            isHiddenNotation(static_cast<KeyUtils::KeyNotation>(newNotation))) {
+        newNotation = newNotation + step;
+        if (newNotation >= numNotations) {
+            newNotation = invalidNotations + 1;
+        } else if (newNotation <= invalidNotations) {
+            newNotation = numNotations - 1;
+        }
+    }
+    m_notation = static_cast<KeyUtils::KeyNotation>(newNotation);
+    // we update the SVG nodes with the new value
+    updateSvg();
+}
+
+void DlgKeywheel::updateSvg() {
+    /* update the svg with new values to display, then cause a an update on the widget */
+    QDomElement topElement = m_domDocument.documentElement();
+
+    bool hideTraditional = m_notation == KeyUtils::KeyNotation::Traditional;
+    QDomElement domElement;
+
+    QDomNodeList node_list = m_domDocument.elementsByTagName(QStringLiteral("text"));
+    for (int i = 0; i < node_list.count(); i++) {
```

Comment:
Can this become a range based loop?

Reply:
no, the QDomNodeList API is very rudimentary.